### PR TITLE
Removes backport and release-notes scripts

### PR DESCRIPTION
**Pull request by tmarkley**
### Description
* Deletes the unnecessary `@osd/release-notes` package.
* Backport is unnecessary since we're adding a backport action in #1233.
* Release notes are generated with `git-release-notes`, and there's no need to maintain a package dedicated to release notes for now. Future work can be done to integrate existing release notes repositories instead of reinventing the wheel.
* Removes unused `@types/graphql` dependency.
 
### Issues Resolved
N/A
 
### Check List

- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by kavilla** [comment]
Do you think we should create a proposal and leave it open for a little? I know within the OpenSearch project we don't utilize existing features but I don't have the data to say forks of OpenSearch Dashboards do not.

I do not believe people are but just to try to get community insight.

**Comment by tmarkley** [comment]
> Do you think we should create a proposal and leave it open for a little? I know within the OpenSearch project we don't utilize existing features but I don't have the data to say forks of OpenSearch Dashboards do not.
> 
> I do not believe people are but just to try to get community insight.

We could, but I don't see why forks would need to use any of this code. Most of it is specific to the Kibana repo anyways, e.g. the tags that they use for release notes.

**Review by ananzh — APPROVED** [review]
LGTM

**Comment by kavilla** [comment]
> Most of it is specific to the Kibana repo anyways, e.g. the tags that they use for release notes. Also, it'd be easy enough to revive this code and add improvements to it in the process.

Makes sense to me.

File: package.json [hunk 1]
```diff
@@ -272,7 +271,6 @@
     "@types/getos": "^3.0.0",
     "@types/glob": "^7.1.3",
     "@types/globby": "^8.0.0",
-    "@types/graphql": "^0.13.2",
```
Comment (kavilla) [inline package.json:275]: Looks like purely an x-pack thing.

**Review by kavilla — APPROVED** [review]
LGTM.

**Pull request merged**